spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceHttpAutoConfiguration.java

```diff
@@ -111,8 +111,9 @@ HttpServerParser defaultHttpServerParser() {
 
 	@Bean
 	@ConditionalOnMissingBean(name = ClientSampler.NAME)
-	HttpSampler sleuthClientSampler() {
-		return HttpSampler.TRACE_ID;
+	HttpSampler sleuthClientSampler(SleuthWebProperties sleuthWebProperties) {
+		return new CompositeHttpSampler(new PathMatchingHttpSampler(
```

Comment:
What is the point in using the composite here since the right always defers? It seems equivalent to just using the `PathMatchingHttpSampler` directly.

Reply:
Oh yeah, I haven't noticed that